src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -569,7 +569,8 @@ private static void checkLiteral(CharSequence string, int index, String literal)
             if (string.charAt(index + i) != literal.charAt(i)) {
```

Comment:
There should probably be a check to verify that `string` contains at least `index + literal.length()` characters.
Otherwise AIOBE might be thrown by string.charAt(index + i) ?

Reply:
The method is only called when the precondition `index + literal.length() <= string.length()`.
I'll add a comment and an assert.